src/relay.cpp

```diff
@@ -479,6 +479,16 @@ void relay_client(int sock, const uint8_t *msg, int32_t len, const ip6_hdr *ip_h
         current_buffer_position += sizeof(ether_hdr->ether_shost);
     }
 
+    if(config->is_interface_id) {
+        interface_id_option intf_id;
+        intf_id.option_code = htons(OPTION_INTERFACE_ID);
+        intf_id.option_length = htons(sizeof(in6_addr));
+        intf_id.interface_id = config->link_address.sin6_addr;
+
+        memcpy(current_buffer_position, &intf_id, sizeof(interface_id_option));
```

Comment:
Does there need to be bounds check added here (and in previous `memcpy` calls to prevent write out-of-bounds? The link MTU is probably 1500 bytes, and the code only picks up one packet at a time (so it won't read a fragmented UDP packet), so it's very unlikely that these calls will fill up the 4096-byte buffer.

Reply:
Added out of bound check